### ISSUE TITLE
anchor-has-content false positive for object spread

### ISSUE THREAD

**Issue by vdh** [evcohen/eslint-plugin-jsx-a11y#87]
A stripped-down version of the component that `anchor-has-content` has an issue with:
```jsx
const Button = ({ className, type, link, title, children }) => {
  const commonProps = { className, title, children };
  if (link) {
    return <Link {...commonProps} to={link} />;
  }
  return <button {...commonProps} type={type} />;
};
```

**Comment by vdh** [comment]
I also ran into a false positive when providing children via prop:
```jsx
<Link to={link} children={children} />
```

**Comment by ljharb** [comment]
imo `children` is special, and you should do:
```jsx
const Button = ({ className, type, link, title, children }) => {
  const commonProps = { className, title, children };
  if (link) {
    return <Link {...commonProps} to={link}>{children}</Link>;
  }
  return <button {...commonProps} type={type}>{children}</button>;
};
```

**Comment by vdh** [comment]
But I thought it was the same way to declare the children, just different notations?
```jsx
import React from 'react';
const Foo = ({ children }) => <div>{children}</div>;
const Bar = ({ children }) => <div children={children} />;
```
Becomes (simplified Babel output):
```js
var React = require('react').default;
var Foo = function Foo(props) {
  return React.createElement('div', null, props.children);
};
var Bar = function Bar(props) {
  return React.createElement('div', { children: props.children });
};
```
And as far as I understand the [React source code](https://github.com/facebook/react/blob/master/src/isomorphic/classic/element/ReactElement.js#L235), both of these arguments will end up being set into the props in the same way? I can't find any documentation warning against providing children in this fashion.

I like to use the prop notation when I'm expecting the child(ren) to be small and uncomplicated (like a label string). And looking at the source code, it also saves React a tiny bit of legwork getting it into the props.

**Comment by ljharb** [comment]
It totally works as you expect. But the linter can't know what props you're spreading. The only options here are:
 - the rule ignores anything using spread props, making it much less useful
 - you disable the rule for this case
 - you put children twice so that the linter sees they are there

The first option greatly reduces the usefulness of the rule imo.

**Comment by vdh** [comment]
In that case it looks like I may get better results by selectively disabling the rule in situations like this, and re-enabling it for the containing component, to pass the responsibility up the chain.

Is it possible to update the rule to allow the `<a children={children} />` notation?

**Comment by ljharb** [comment]
Oh, it should absolutely be covering an explicit "children" prop! Your original example didn't have that - does it still warn if you have `<Link {...commonProps} children={children} to={link} />`? If so, that's a bug!
